sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -325,20 +295,10 @@ abstract class AbstractGpuCoalesceIterator(
               saveOnDeck(cb)
             } else if (wouldBeBytes > goal.targetSizeBytes && numBytes > 0) {
               saveOnDeck(cb)
```

Comment:
nit: Might be nice to have a comment here about why we don't need to worry about overflow for strings/etc.  Might also be nice to have an assertion some where, perhaps in `TagetSize` itself to verify we did not make a mistake and go over the goal.  Just trying to be defensive in this.

Reply:
We cannot put in an assert on `TargetSize` since `RequireSingleBatch` forces us to coalesce above the 2GB goal if necessary.  If we cannot rely on the cudf C++ exception to cover the case where we exceed the cudf row limit then we'll need to continue checking individual column lengths.

Reply:
I should clarify that we can put an assert on `TargetSize`, but the code has to handle the case where we're above 2GB due to `RequireSingleBatch`.

Reply:
Yes, I know which is why I wanted the comments and the assertion.  The comments are to explain to someone in the future that we are relying on CUDF concat on `RequireSingleBatch` to do the right thing and blow up if any of the child columns overflow, and that we are limiting the `TargetSize` to 2GB so there is no possibility of an overflow in that case. All of this is a nit anyways I am just trying to be defensive and if we have an assumption try and enforce that assumption and error out as early as possible.

Reply:
I added a `TargetSize` check and some comments.